### PR TITLE
migrate to test app

### PR DESCRIPTION
**Pull request by paulomach**
## Issue

Usage of mysql-test-app all across our charm tests, replacing test applications within the test code.
Tests that use 3rd party apps for testing should remain the same.

Part of [DPE-1465](https://warthogs.atlassian.net/browse/DPE-1465)

## Solution

Replaced test application for relations and HA tests with mysql-test-app

[DPE-1465]: https://warthogs.atlassian.net/browse/DPE-1465?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by taurus-forever** [comment]
@paulomach I think the test is failing as mysql-interface-database+mysql-interface-user are missing
[here](https://github.com/canonical/mysql-operator/pull/159/files#diff-9fd29972a6d77f9c9e7d6123ab9842892f51a875809992e46381378d3f9ddb04R53)

It is reported [here](https://github.com/canonical/mysql-test-app/pull/4#pullrequestreview-1354106084).

**Review by taurus-forever — CHANGES_REQUESTED** [review]

File: tests/integration/relations/test_relation_mysql_legacy.py [hunk 1]
```diff
@@ -56,7 +51,12 @@ async def test_build_and_deploy(ops_test: OpsTest, mysql_charm_series: str) -> N
             num_units=3,
             series=mysql_charm_series,
         ),
-        ops_test.model.deploy(app_charm, application_name=APPLICATION_APP_NAME, num_units=2),
+        ops_test.model.deploy(
+            APPLICATION_APP_NAME,
+            application_name=APPLICATION_APP_NAME,
+            num_units=2,
+            channel="latest/edge",  # TODO: Remove branch before merge
```
[the remark this answers is not in the document]
Comment (paulomach) [reply tests/integration/relations/test_relation_mysql_legacy.py:58]: ops - I released is a test branch `latest/edge/legacy-mysql` for the test charm, but  `pytest_operator` (or libjuju) is not able to deploy it (forgot to remove the comment)

**Comment by paulomach** [comment]
> @paulomach I think the test is failing due to [lack of config options](https://github.com/canonical/mysql-operator/pull/159/files#diff-9fd29972a6d77f9c9e7d6123ab9842892f51a875809992e46381378d3f9ddb04R53) `mysql-interface-database`+`mysql-interface-user`
> 
> It is reported [here](https://github.com/canonical/mysql-test-app/pull/4#pullrequestreview-1354106084).

Not quite. Test app is being deployed with two units, and non leader unit is not setting Active state, holding app state in `waiting`. Addressed on [test app](https://github.com/canonical/mysql-test-app/commit/587e7a890743d98f9450fb56d5b1b4be99d433e7)

**Review by taurus-forever — APPROVED** [review]
LGTM. Please remove TODO comment.

**Pull request merged**